extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -144,6 +159,18 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   attachContent_() {
+    // Set quiz ID
+    if (!this.element.hasAttribute('id')) {
+      // If the quiz has no ID, randomly set one
```

Comment:
This could potentially give two quizzes the same id. You can just have a counter and increment it each time you give out an id.

Reply:
Changed this so instead it relies on the publisher supplying an ID, which will be enforced with validation